packages/cli/generators/datasource/index.js

```diff
@@ -0,0 +1,290 @@
+   */
+  promptConnectorConfig() {
+    debug('prompting for connector config');
+    // Check to make sure connector is from connectors list (not custom)
+    const settings = connectors[this.artifactInfo.connector]
+      ? connectors[this.artifactInfo.connector]['settings']
+      : {};
+
+    const prompts = [];
+    // Create list of questions to prompt the user
+    Object.entries(settings).forEach(([key, setting]) => {
+      // Set defaults and merge with `setting` to override properties
+      const question = Object.assign(
+        {},
+        {name: key, message: key, suffix: ':'},
+        setting,
+      );
+
+      /**
+       * Allowed Types: string, number, password, object, array, boolean
+       * Must be converted to inquirer types -- input, confirm, password
+       */
+      switch ((setting.type || '').toLowerCase()) {
+        case 'string':
+        case 'number':
+          question.type = 'input';
+          break;
+        case 'object':
+        case 'array':
+          question.type = 'input';
+          question.validate = utils.validateStringObject(setting.type);
+          break;
+        case 'boolean':
+          question.type = 'confirm';
+          break;
+        case 'password':
+          break;
+        default:
+          console.warn(
+            `Using default input of type input for setting ${key} as ${setting.type ||
+              undefined} is not supported`,
+          );
+          // Default to input type
+          question.type = 'input';
+      }
+
+      prompts.push(question);
+    });
+
+    debug(`connector setting questions - ${JSON.stringify(prompts)}`);
+
+    // If no prompts, we need to return instead of attempting to ask prompts
+    if (!prompts.length) return;
+
+    debug('prompting the user - length > 0 for questions');
+    // Ask user for prompts
+    return this.prompt(prompts).then(props => {
+      // Convert user inputs to correct types
+      Object.entries(settings).forEach(([key, setting]) => {
+        switch ((setting.type || '').toLowerCase()) {
+          case 'number':
+            props[key] = Number(props[key]);
+            break;
+          case 'array':
+          case 'object':
+            if (props[key] == null || props[key] === '') {
+              delete props[key];
+            } else {
+              props[key] = JSON.parse(props[key]);
+            }
+            break;
+        }
+      });
+      this.artifactInfo = Object.assign(this.artifactInfo, {settings: props});
+    });
+  }
+
+  install() {
+    debug('install npm dependencies');
+    const pkgs = [];
+
+    // Connector package.
+    const connector = connectors[this.artifactInfo.connector];
+    if (connector && connector.package) {
+      pkgs.push(
+        connector.package.name +
+          `${connector.package.version ? '@' + connector.package.version : ''}`,
+      );
+
+      debug(`npmModule - ${pkgs[0]}`);
+    }
+
+    pkgs.push('@loopback/repository');
```

Comment:
would it make sense to install `@loopback/core` here as well since we import `inject` from it in the template datasource ejs file?

Reply:
This command is meant to be run inside a loopback 4 project which if generated using CLI should already have `@loopback/core` installed but we don't install `@loopback/repository`.